README.md

```diff
@@ -8,3 +8,53 @@
+
+To build the API-Server `cd` to this repository and enter the following command: 
+
+```
+docker build -t csblogs-api-server .
+```
+
+This builds the docker image and allows you to refer to it later with the tag `csblogs-api-server`. To then run an container instance of the API-Server you can issue the following command:
+
+```
+docker run -p 80:80 csblogs-api-server
+```
+
+The `-p` flag maps port 80 on your machine to port 80 inside the container. The API-Server binds to port 80 inside the container. If you'd rather have it on another port on your host machine change the value on the left. `3000:80` would allow you to access the API-Server on port 3000 from your machine.
+
+You'll likely notice at this point that the application logs an error:
+
+```
+Connection to Database Failed { host: undefined,
+      port: undefined,
+      name: undefined,
+      username: undefined }
+```
+
+This is because the connection information for the database is provided through environment variables which we haven't set. You can set each of these environment variables using the `-e` flag to the `docker run` command. The environment variable names are listed in the `docker-compose.yml` file. Clearly, in order to be able to do this you will need your own PostgreSQL database. This is where `docker-compose` comes in -- it allows us to stand up both a PostgreSQL container and an API-Server container using just one command and allows us to set environment variables in a file, rather than having to have many `-e` flags in our `docker run` command.
+
+### Docker compose
+Before you can use `docker-compose` you need to add values for each of the environment variables in the `docker-compose.yml` file. The `POSTGRES_USER` and `POSTGRES_PASSWORD` should match up with `CSBLOGS_DATABASE_USERNAME` and `CSBLOGS_DATABASE_PASSWORD` respectively. `CSBLOGS_DATABASE_PORT` needs to match postgres, which is usually `5432` and the hostname of the Postgres container will be the service name provided -- in this case that's `csblogs-postgres`.
+
+Once you have added the environment variable properties you can simple run `docker-compose up` to get both an API Server and the supporting database.
+
+## Testing
+The API-Server comes with a bunch of tests written in a BDD fashion using `chai` and `mocha`. You can run these in your docker container by issuing the command `docker run csblogs-api-server test`.
+
+## Linting
+All code for the CSBlogs project uses the Airbnb `ESLint` style rules. To run the linter against your code in order to check if it meets the projects style guidlines you can issue the command `docker run csblogs-api-server lint`.
+
+## Deploying
+The API-Server is hosted on [Zeit Now](https://zeit.co/now). One of the cool things about using `Now` is that anyone can deploy, up to 20 times a month, for free. So to have your own instance of the API-Server running in the cloud (at no cost) you just have to issue a few commands.
```

Comment:
"is that anyone can deploy, up to 20 times a month" - you should clarify each developer has 20 free deployments a month, otherwise it could be construed that anyone can deploy but that the repo can only have 20 deployments a month